tests/components/iaqualink/test_init.py

```diff
@@ -0,0 +1,360 @@
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # False -> None / ServiceException
+    system.online = False
+    with patch(
+        "iaqualink.system.AqualinkSystem.update",
+        side_effect=AqualinkServiceException,
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # None -> None / ServiceException
+    system.online = None
+    with patch(
+        "iaqualink.system.AqualinkSystem.update",
+        side_effect=AqualinkServiceException,
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # None -> True
+    system.online = None
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_true
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_called_once()
+
+    # None -> False
+    system.online = None
+    with patch(
+        "iaqualink.system.AqualinkSystem.update", side_effect=set_online_to_false
+    ), patch("homeassistant.components.iaqualink._LOGGER.warning") as mock_warn:
+        _ffwd_next_update_interval(hass)
+        await hass.async_block_till_done()
+        mock_warn.assert_not_called()
+
+    assert await hass.config_entries.async_unload(config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert DOMAIN not in hass.data
+
+
+async def test_entity_assumed_and_available(hass):
+    """Test assumed_state and_available properties for all values of online."""
+    client = get_aqualink_client()
+    system = get_aqualink_system(client)
+    light = get_aqualink_device(system, AqualinkLightToggle)
+    ha_light = AqualinkEntity(light)
+
+    # None means maybe.
+    light.system.online = None
+    assert ha_light.assumed_state is True
+    assert ha_light.available is False
+
+    light.system.online = False
+    assert ha_light.assumed_state is True
+    assert ha_light.available is False
+
+    light.system.online = True
+    assert ha_light.assumed_state is False
+    assert ha_light.available is True
+
+
+async def test_try_await(hass):
+    """Test try_await for all values of awaitable."""
+    async_noop = async_returns(None)
+    await try_await(async_noop())
+
+    with pytest.raises(Exception):
+        async_ex = async_raises(Exception)
+        await try_await(async_ex())
+
+    with pytest.raises(HomeAssistantError):
+        async_ex = async_raises(AqualinkServiceException)
+        await try_await(async_ex())
```

Comment:
Usually we try not to interact with any of the integration code in tests https://developers.home-assistant.io/docs/development_testing/#writing-tests-for-integrations

If you want to test a utility function directly, move it into a `utils` module